builtin/providers/aws/resource_aws_route53_zone.go

```diff
@@ -90,6 +96,8 @@ func resourceAwsRoute53ZoneRead(d *schema.ResourceData, meta interface{}) error
 		return err
 	}
 
+	d.Set("delegation_set_name_servers", zone.DelegationSet.NameServers)
```

Comment:
`d.Set` can return an error, can you check to make sure it's `nil`? We've seen bugs go by silently by not checking for errors when setting a non-primitive type. 

`if err := d.Set("delegation_set_name_servers", zone.DelegationSet.NameServers); err != nil{`

or similar, like in https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_route53_record.go#L199